play/server/src/main/scala/endpoints/play/server/JsonEntities.scala

```diff
@@ -71,14 +72,32 @@ private object JsonEntities {
 
   def decodeRequest[A](
       endpoints: EndpointsWithCustomErrors
-  )(decoder: Decoder[String, A]): BodyParser[A] =
-    endpoints.playComponents.playBodyParsers.byteString.validate { bs =>
-      decoder
-        .decode(bs.utf8String)
-        .toEither
-        .left
-        .map(errs => endpoints.handleClientErrors(Invalid(errs)))
-    }(endpoints.playComponents.executionContext)
+  )(decoder: Decoder[String, A]): BodyParser[A] = new BodyParser[A] {
+    def apply(request: RequestHeader) = {
+      if (request.contentType.exists(_.equalsIgnoreCase("application/json"))) {
+        val decodeJson = (bs: ByteString) =>
+          decoder
+            .decode(bs.utf8String)
+            .toEither
+            .left
+            .map(errs => endpoints.handleClientErrors(Invalid(errs)))
+
+        endpoints.playComponents.playBodyParsers.byteString
+          .validate(decodeJson)(endpoints.playComponents.executionContext)
+          .apply(request)
+      } else {
+        Accumulator.done {
+          endpoints.playComponents.httpErrorHandler
+            .onClientError(
+              request,
+              Status.UNSUPPORTED_MEDIA_TYPE,
+              "Expecting application/json body"
+            )
+            .map(Left(_))(endpoints.playComponents.executionContext)
+        }
```

Comment:
We should follow the same pattern as with bad request errors and return a response ourselves rather than forwarding to the `httpErrorHandler`.

Reply:
See the comment about the Akka HTTP case. Since I don't use Play, I don't care much about the decision here, but I expect that Play users would at least care about uniformity/customizability of 415 responses (especially since customizing 415's looks like a feature - https://www.playframework.com/documentation/2.8.x/ScalaErrorHandling#Extending-the-default-error-handler looks like something I would want to be able to use).